teuthology/repo_utils.py

```diff
@@ -368,31 +380,33 @@ def url_to_dirname(url):
     return string
 
 
-def fetch_qa_suite(branch, lock=True):
+def fetch_qa_suite(branch, commit=None, lock=True):
     """
     Make sure ceph-qa-suite is checked out.
 
     :param branch: The branch to fetch
+    :param commit: The commit to checkout
     :returns:      The destination path
     """
     return fetch_repo(config.get_ceph_qa_suite_git_url(),
-                      branch, lock=lock)
+                      branch, commit, lock=lock)
 
 
-def fetch_teuthology(branch, lock=True):
+def fetch_teuthology(branch, commit=None, lock=True):
     """
     Make sure we have the correct teuthology branch checked out and up-to-date
 
     :param branch: The branch we want
+    :param commit: The commit to checkout
     :returns:      The destination path
     """
     url = config.ceph_git_base_url + 'teuthology.git'
-    return fetch_repo(url, branch, bootstrap_teuthology, lock)
+    return fetch_repo(url, branch, commit, bootstrap_teuthology, lock)
 
 
-def bootstrap_teuthology(dest_path):
+def bootstrap_teuthology(dest_path, commit):
```

Comment:
It is useless to have commit for the bootstrap method (especially without branch context), so check commit is empty on the caller side.